src/connection-pool.js

```diff
@@ -200,10 +219,13 @@ ConnectionPool.prototype.createConnection = function() {
       .once(CHANNEL_ERROR_EVENT, onChannelError)
       .once(CHANNEL_READY_EVENT, onChannelReady);
 
+    requestStream.on('status', function(status) {
+      setImmediate(() => onConnectionStatus(status));
```

Comment:
Can you add a code comment to explain why we need this?

Also, I think `setImmediate(onConnectionStatus, status)` works.